src/rapids_pre_commit_hooks/copyright.py

```diff
@@ -0,0 +1,213 @@
+        except IndexError:
+            pass
+    target_branch_name = os.getenv("TARGET_BRANCH")
+    if target_branch_name:
+        try:
+            return repo.heads[target_branch_name]
+        except IndexError:
+            pass
+    target_branch_name = os.getenv("RAPIDS_BASE_BRANCH")
+    if target_branch_name:
+        try:
+            return repo.heads[target_branch_name]
+        except IndexError:
+            pass
+
+    # Try config
+    with repo.config_reader() as r:
+        target_branch_name = r.get("rapidsai", "baseBranch", fallback=None)
+    if target_branch_name:
+        try:
+            return repo.heads[target_branch_name]
+        except IndexError:
+            pass
+
+    # Try newest branch-xx.yy
+    branches = sorted(
+        (
+            (branch, (match.group("major"), match.group("minor")))
+            for branch in repo.heads
+            if (match := BRANCH_RE.search(branch.name))
+        ),
+        key=lambda i: i[1],
+        reverse=True,
+    )
+    try:
+        return branches[0][0]
+    except IndexError:
+        pass
+
+    # Appropriate branch not found
+    return None
+
+
+def get_target_branch_upstream_commit(repo):
+    target_branch = get_target_branch(repo)
+    if target_branch is None:
+        return repo.head.commit
+
+    target_branch_upstream = target_branch.tracking_branch()
+    if target_branch_upstream:
+        return target_branch_upstream.commit
+
+    def try_get_ref(remote):
+        try:
+            return remote.refs[target_branch.name]
+        except IndexError:
+            return None
+
+    candidate_upstreams = sorted(
+        (upstream for remote in repo.remotes if (upstream := try_get_ref(remote))),
+        key=lambda upstream: upstream.commit.committed_datetime,
+        reverse=True,
+    )
+    try:
+        return candidate_upstreams[0].commit
+    except IndexError:
+        pass
+
+    return target_branch.commit
+
+
+def get_changed_files(repo, target_branch_upstream_commit):
+    changed_files = {}
+
+    diffs = target_branch_upstream_commit.diff(
+        other=None,
+        merge_base=True,
+        find_copies=True,
+        find_copies_harder=True,
+        find_renames=True,
+    )
+    for diff in diffs:
+        if diff.change_type == "A":
+            changed_files[diff.b_path] = None
+        elif diff.change_type != "D":
+            changed_files[diff.b_path] = diff.a_blob
+
+    changed_files.update({f: None for f in repo.untracked_files})
+    return changed_files
+
+
+def check_copyright():
+    repo = git.Repo()
+    target_branch_upstream_commit = get_target_branch_upstream_commit(repo)
+    changed_files = get_changed_files(repo, target_branch_upstream_commit)
+
+    def the_check(linter, args):
+        try:
+            changed_file = changed_files[linter.filename]
+        except KeyError:
+            return
+
+        old_content = changed_file.data_stream.read().decode("utf-8")
```

Comment:
I think utf-8 is the default and could be omitted.
```suggestion
        old_content = changed_file.data_stream.read().decode()
```